packages/radio-player/src/search-handler/search-handler.ts

```diff
@@ -0,0 +1,147 @@
+        const intersection = SearchHelper.getIntersection(entry.range, indexMap.range);
+        if (!intersection || intersection.length === 0) {
+          return;
+        }
+
+        const newTranscriptEntry = this.createBlankTranscriptEntryConfig(indexMap.entry);
+        const text = this.searchIndex.mergedTranscript.substring(
+          intersection.startIndex,
+          intersection.endIndex,
+        );
+        newTranscriptEntry.rawText = text.trim();
+        newTranscriptEntries.push(newTranscriptEntry);
+      });
+    });
+
+    const newTranscript = new TranscriptConfig(newTranscriptEntries);
+
+    return newTranscript;
+  }
+
+  /**
+   * Copy a transcript entry but leave the text and search result index empty.
+   *
+   * @private
+   * @param {TranscriptEntryConfig} sourceTranscriptConfig
+   * @returns {TranscriptEntryConfig}
+   * @memberof SearchHandler
+   */
+  /* eslint-disable-next-line class-methods-use-this */
+  private createBlankTranscriptEntryConfig(
+    sourceTranscriptConfig: TranscriptEntryConfig,
+  ): TranscriptEntryConfig {
+    return new TranscriptEntryConfig(
+      sourceTranscriptConfig.id,
+      sourceTranscriptConfig.start,
+      sourceTranscriptConfig.end,
+      '',
+      sourceTranscriptConfig.isMusic,
+    );
+  }
+
+  /**
+   * Search the full transcript and split up by search results and non-results. For instance,
+   * if the full transcript is `foo bar baz boop bump snap pop` and you search for `bump`,
+   * you'll get an array of 3 results back:
+   * 1. `foo bar baz boop `
+   * 2. `bump` <-- the match
+   * 3. ` snap pop`
+   *
+   * This is helpful when rebuilding the transcript later to be able to identify search results.
+   *
+   * @private
+   * @param {string} term
+   * @returns {SearchResult[]}
+   * @memberof SearchHandler
+   */
+  private getSearchSeparatedTranscript(term: string): SearchResult[] {
+    const searchRanges: Range[] = this.searchIndex.getSearchRanges(term);
+    const { mergedTranscript } = this.searchIndex;
+
+    if (searchRanges.length === 0) {
+      const range = new Range(0, mergedTranscript.length);
+      return [new SearchResult(range, mergedTranscript, false)];
+    }
+
+    const transcriptEntries: SearchResult[] = [];
+    let startIndex = 0;
+    searchRanges.forEach(range => {
+      const nextStart = range.endIndex;
+      const nonResultText = mergedTranscript.substring(startIndex, range.startIndex);
+      const resultText = mergedTranscript.substring(range.startIndex, nextStart);
+      const nonResultRange = new Range(startIndex, range.startIndex - 1);
+      const nonResultEntry = new SearchResult(nonResultRange, nonResultText, false);
+      const searchResultRange = new Range(range.startIndex, nextStart - 1);
+      const searchResultEntry = new SearchResult(searchResultRange, resultText, true);
+      transcriptEntries.push(nonResultEntry);
+      transcriptEntries.push(searchResultEntry);
+      startIndex = nextStart;
+    });
```

Comment:
I might just need a walk through this part with you.

Reply:
I cleaned up this part and added some documentation as to what's going on through it. Take another look and let me know if it's more clear now.